File: app/src/main/java/io/trewartha/positional/position/CoordinatesUTMFragment.java
package io.trewartha.positional.position;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.Locale;

import gov.nasa.worldwind.geom.Angle;
import gov.nasa.worldwind.geom.coords.UTMCoord;
import io.trewartha.positional.R;

public class CoordinatesUTMFragment extends CoordinatesFragment {

    private static final String FORMAT = "%7.0f";
    private static final Locale LOCALE = Locale.getDefault();

    private TextView zoneTextView;
    private TextView eastingTextView;
    private TextView northingTextView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        final View view = inflater.inflate(R.layout.coordinates_utm_fragment, container, false);
        zoneTextView = view.findViewById(R.id.utm_zone_text_view);
        eastingTextView = view.findViewById(R.id.utm_easting_text_view);
        northingTextView = view.findViewById(R.id.utm_northing_text_view);
        setCoordinates(latitude, longitude);
        return view;
    }

    @Override
    public void setCoordinates(double latitude, double longitude) {
        if (zoneTextView != null && eastingTextView != null && northingTextView != null) {
            final UTMCoord utmCoord = UTMCoord.fromLatLon(
                    Angle.fromDegreesLatitude(latitude),
                    Angle.fromDegreesLongitude(longitude)
            );
            final String zone = utmCoord.getZone() + ("gov.nasa.worldwind.avkey.North".equals(utmCoord.getHemisphere()) ? "N " : "S ");
            final String easting = String.format(LOCALE, FORMAT, utmCoord.getEasting()) + "m E";
            final String northing = String.format(LOCALE, FORMAT, utmCoord.getNorthing()) + "m N";

            zoneTextView.setText(zone);
            eastingTextView.setText(easting);
            northingTextView.setText(northing);
        }
    }
}

File: app/src/main/java/io/trewartha/positional/position/CoordinatesUTMFragment.kt
package io.trewartha.positional.position

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import io.trewartha.positional.R
import kotlinx.android.synthetic.main.coordinates_utm_fragment.*

class CoordinatesUTMFragment : CoordinatesFragment() {

    private lateinit var locationFormatter: LocationFormatter

    override fun onAttach(context: Context) {
        super.onAttach(context)
        locationFormatter = LocationFormatter(context)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.coordinates_utm_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setCoordinates(latitude, longitude)
    }

    override fun setCoordinates(latitude: Double, longitude: Double) {
        zoneTextView?.text = locationFormatter.getUtmZone(latitude, longitude)
        eastingTextView?.text = locationFormatter.getUtmEasting(latitude, longitude)
        northingTextView?.text = locationFormatter.getUtmNorthing(latitude, longitude)
    }
}